core/src/main/java/org/apache/accumulo/core/client/summary/ColumnVisibilitySummarizer.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.accumulo.core.client.summary;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.commons.lang.mutable.MutableLong;
+
+public class ColumnVisibilitySummarizer implements KeyValueSummarizer {
+
+  private static final int MAX = 5000;
+  private static final String PREFIX = "cv:";
+  private static final String IGNORE_KEY = "ignored";
+
+  // Map used for computing summary incrementally uses ByteSequence for key which is more efficient than converting colvis to String for each Key. The
+  // conversion to String is deferred until the summary is requested.  This shows how the interface enables users to write efficient summarizers.
+  private Map<ByteSequence,MutableLong> summary = new HashMap<>();
+  private long ignored = 0;
+
+  @Override
+  public String getId() {
+    return "accumulo.cvsummary";
+  }
+
+  @Override
+  public void collect(Key k, Value v) {
```

Comment:
Yeah, my thinking is that every key/value added to the RFile would also be passed to this function